src/camera_line_follower_component.cpp

```diff
@@ -0,0 +1,272 @@
+// Copyright 2020 RT Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "raspimouse_ros2_examples/camera_line_follower_component.hpp"
+
+#include <memory>
+#include <chrono>
+#include <iostream>
+#include <utility>
+#include <string>
+#include <vector>
+
+#include <opencv2/opencv.hpp>
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+#include "lifecycle_msgs/srv/change_state.hpp"
+#include "cv_bridge/cv_bridge.h"
+
+using namespace std::chrono_literals;
+using CallbackReturn = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn;
+
+namespace camera_line_follower
+{
+
+Camera_Follower::Camera_Follower(const rclcpp::NodeOptions & options)
+: rclcpp_lifecycle::LifecycleNode("camera_follower", options),
+  object_is_detected_(false)
+{
+}
+
+void Camera_Follower::image_callback(const sensor_msgs::msg::Image::SharedPtr msg_image)
+{
+  auto cv_img = cv_bridge::toCvShare(msg_image, msg_image->encoding);
+  auto msg = std::make_unique<sensor_msgs::msg::Image>();
+  auto result_msg = std::make_unique<sensor_msgs::msg::Image>();
+  msg->is_bigendian = false;
+  result_msg->is_bigendian = false;
+
+  cv::Mat frame, result_frame;
+  cv::cvtColor(cv_img->image, frame, CV_RGB2BGR);
+
+  if (!frame.empty()) {
+    following(frame, result_frame);
+    convert_frame_to_message(result_frame, *result_msg);
+    result_image_pub_->publish(std::move(result_msg));
+  }
+}
+
+void Camera_Follower::callback_switches(const raspimouse_msgs::msg::Switches::SharedPtr msg)
+{
+  switches_ = *msg;
+}
+
+void Camera_Follower::on_cmd_vel_timer()
+{
+  const double LINEAR_VEL = 0.2;  // unit: m/s
+  const double ANGULAR_VEL = -0.8;  // unit: rad/s
+  const double TARGET_AREA = 0.1;  // 0.0 ~ 1.0
+  const double OBJECT_AREA_THRESHOLD = 0.01;  // 0.0 ~ 1.0
```

Comment:
コンパイル時に定数化できるのでconstexprを使いましょう。

```suggestion
  constexpr double LINEAR_VEL = 0.2;  // unit: m/s
  constexpr double ANGULAR_VEL = -0.8;  // unit: rad/s
  constexpr double TARGET_AREA = 0.1;  // 0.0 ~ 1.0
  constexpr double OBJECT_AREA_THRESHOLD = 0.01;  // 0.0 ~ 1.0
```